testing/regress/ecl/phonenumber.ecl

```diff
@@ -0,0 +1,193 @@
+  
+  // Numbers with wrong country codes
+  {'+1 20 7946 0958', 'GB', false, 'UK number with US country code'},
+  {'+44 516 924 2448', 'US', false, 'US number with UK country code'},
+  {'+49 3 1234 5678', 'JP', false, 'Japanese number with German country code'},
+  {'+81 30 123456', 'DE', false, 'German number with Japanese country code'},
+  
+  // Numbers without '+' prefix - Valid international format without plus
+  {'1 516 924 2448', 'US', true, 'US number without plus prefix'},
+  {'44 20 7946 0958', 'GB', true, 'UK London number without plus prefix'},
+  {'49 30 123456', 'DE', true, 'German Berlin number without plus prefix'},
+  {'81 3 1234 5678', 'JP', true, 'Japanese Tokyo number without plus prefix'},
+  {'86 10 1234 5678', 'CN', true, 'Chinese Beijing number without plus prefix'},
+  {'33 1 42 68 53 00', 'FR', true, 'French Paris number without plus prefix'},
+  {'39 06 698 80000', 'IT', true, 'Italian Rome number without plus prefix'},
+  
+  // Numbers without '+' prefix - National format
+  {'20 7946 0958', 'GB', true, 'UK national format without country code'},
+  {'30 123456', 'DE', true, 'German national format without country code'},
+  {'3 1234 5678', 'JP', true, 'Japanese national format without country code'},
+  {'10 1234 5678', 'CN', true, 'Chinese national format without country code'},
+  {'1 42 68 53 00', 'FR', true, 'French national format without country code'},
+  {'06 698 80000', 'IT', true, 'Italian national format without country code'},
+  
+  // Numbers without '+' prefix - Invalid cases
+  {'1 555 012 3456', 'US', false, 'US invalid number without plus prefix'},
+  {'44 20 123', 'GB', false, 'UK too short without plus prefix'},
+  {'49 30 123', 'DE', false, 'German too short without plus prefix'},
+  {'81 3 123', 'JP', false, 'Japanese too short without plus prefix'},
+  {'86 10 123', 'CN', false, 'Chinese too short without plus prefix'},
+  {'33 1 42', 'FR', false, 'French too short without plus prefix'},
+  {'39 06 698', 'IT', false, 'Italian too short without plus prefix'},
+  
+  // Domestic format numbers (no country code at all)
+  {'7946 0958', 'GB', true, 'UK domestic format'},
+  {'123456', 'DE', true, 'German domestic format'},
+  {'1234 5678', 'JP', true, 'Japanese domestic format'},
+  {'1234 5678', 'CN', true, 'Chinese domestic format'},
+  {'42 68 53 00', 'FR', true, 'French domestic format'},
+  {'698 80000', 'IT', true, 'Italian domestic format'},
+  
+  // Edge cases - numbers that could be ambiguous
+  {'123456789', 'US', false, 'Ambiguous 9-digit number'},
+  {'1234567890', 'US', true, 'US 10-digit domestic format'},
+  {'01234567890', 'GB', true, 'UK with leading zero'},
+  {'0123456789', 'DE', true, 'German with leading zero'},
+  {'03 1234 5678', 'JP', true, 'Japanese with leading zero'},
+  {'010 1234 5678', 'CN', true, 'Chinese with leading zero'},
+  {'01 42 68 53 00', 'FR', true, 'French with leading zero'},
+  {'06 698 80000', 'IT', true, 'Italian with leading zero'}
```

Comment:
I would add some tests where the two letter country code is omitted?

Reply:
Good idea.